# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLBlitHelper.cpp
# Commit: f40ca5c464c2
# Full Hash: f40ca5c464c2811790081dd4c41f0388a7c422aa
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2021-01-20 09:51:21
# Description:
#   Bug 1687438 - Bail if null iosurf for GLBlitHelper::BlitImage. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D102373
# ==============================================================================

diff -r 113beccee269 -r f40ca5c464c2 gfx/gl/GLBlitHelper.cpp
--- a/gfx/gl/GLBlitHelper.cpp	Wed Jan 20 01:35:31 2021 +0000
+++ b/gfx/gl/GLBlitHelper.cpp	Wed Jan 20 01:49:32 2021 +0000
@@ -982,6 +982,10 @@
 bool GLBlitHelper::BlitImage(MacIOSurface* const iosurf,
                              const gfx::IntSize& destSize,
                              const OriginPos destOrigin) const {
+  if (!iosurf) {
+    gfxCriticalError() << "Null MacIOSurface for GLBlitHelper::BlitImage";
+    return false;
+  }
   if (mGL->GetContextType() != GLContextType::CGL) {
     MOZ_ASSERT(false);
     return false;
